packages/forms/components/TextInput.tsx

```diff
@@ -0,0 +1,171 @@
+   */
+  inputLabel: string | React.ReactNode;
+  /**
+   * Defaults to `true`, but can be set to `false` to visibly hide the `TextInput`'s label. The `inputLabel` should still be set even when hidden for accessibility support.
+   */
+  showInputLabel: boolean;
+  /**
+   * Sets the contents for a validation error, this can be a `string` or any `ReactNode`. This will be displayed below the input element. validationContent is only visible when the `TextInput` appearance is also set to `TextInputAppearance.Error`.
+   */
+  validationContent?: string | React.ReactNode;
+}
+
+export class TextInput<
+  P extends TextInputProps,
+  S extends {}
+> extends React.PureComponent<P, S> {
+  public static defaultProps: Partial<TextInputProps> = {
+    type: "text",
+    appearance: TextInputAppearance.Standard,
+    showInputLabel: true
+  };
+
+  public render() {
+    const labelContent = this.getLabelContent();
+    const containerProps: { className?: string } = {};
+    if (this.props.className) {
+      containerProps.className = this.props.className;
+    }
+    return (
+      <div {...containerProps}>
+        {labelContent}
+        {this.getInputContent()}
+        {this.getValidationContent()}
+      </div>
+    );
+  }
+
+  protected getInputAppearance(): string {
+    return this.props.disabled ? "disabled" : this.props.appearance;
+  }
+
+  protected getLabelContent() {
+    const labelClassName = this.props.showInputLabel
+      ? cx(
+          flush("top"),
+          margin("bottom", "xxs"),
+          textWeight("medium"),
+          display("block"),
+          {
+            [tintText(errorColor)]:
+              this.props.appearance === TextInputAppearance.Error
+          }
+        )
+      : cx(visuallyHidden);
+    return (
+      <label className={labelClassName} htmlFor={this.props.id}>
+        {this.props.inputLabel}
+      </label>
+    );
+  }
+
+  protected getInputContent(): React.ReactNode {
+    return (
+      <span>
+        {this.getInputElement([
+          padding("left", "m"),
+          padding("right", "m"),
+          inputContainer,
+          inputAppearances[this.getInputAppearance()]
+        ])}
+      </span>
+    );
+  }
+  protected getInputElementProps() {
+    // omit props for container and that we override, otherwise pass through
+    // TextInput props to input element
+    const {
+      appearance,
+      className,
+      inputLabel,
+      showInputLabel,
+      type,
+      validationContent,
+      ...inputElementProps
+    } = this.props as TextInputProps;
+    if (appearance === TextInputAppearance.Error) {
+      inputElementProps["aria-invalid"] = true;
+      if (validationContent) {
+        inputElementProps["aria-describedby"] = "errorMsg";
+      }
+    }
+    return inputElementProps;
+  }
+
+  protected getInputElement(additionalClasses: string[] = []) {
+    const inputElementProps = this.getInputElementProps();
+
+    return (
+      <input
```

Comment:
You're getting React props in your HTML:
![screen shot 2018-12-05 at 12 55 08 pm](https://user-images.githubusercontent.com/2313998/49536762-47031500-f895-11e8-91b9-dd8c805417d6.png)

Reply:
Doh! I need to omit the icon pros in `TextInputWithIcon`